src/main.rs

```diff
@@ -73,52 +77,53 @@ fn three_letter_code(aa: char) -> String {
     ]
     .iter()
     .copied()
-    .collect();
+    .collect()
+});
 
+/// get the three-letter equivalent of an amino acid
+/// aa (char): the amino acid to translate, eg 'A' -> "Ala"
+/// returns: String
+fn three_letter_code(aa: char) -> Result<String, Error> {
     // check input.
-    let re = Regex::new(r"[A-Z*]").unwrap();
-    if !re.is_match(&aa.to_string()) {
-        println!("Invalid amino acid: {}", aa);
-        process::exit(1);
+    match aa {
+        'A'..='Z' => Ok(THREE_LETTER_CODE[&aa].to_string()),
+        _ => Err(Error::InvalidAminoAcid(aa)),
     }
-    three_letter_map[&aa].to_string()
 }
 
-// translate a codon into its corresponding amino acid
-// triplet (&str): a three-letter codon eg "ATG"
-// i (usize): codon position. if 0, use the STARTS table
-// returns: char
-fn translate(triplet: &str, i: usize) -> char {
+/// translate a codon into its corresponding amino acid
+/// triplet (&str): a three-letter codon eg "ATG"
+/// i (usize): codon position. if 0, use the STARTS table
+/// returns: char
+fn translate(triplet: &[u8], i: usize) -> Result<char, Error> {
```

Comment:
Again changing from `&str` to `&[u8]`, and now also returning possible errors with `Result<char, Error>`